crates/router/src/db/business_profile.rs

```diff
@@ -114,38 +114,118 @@ impl BusinessProfileInterface for Store {
 impl BusinessProfileInterface for MockDb {
     async fn insert_business_profile(
         &self,
-        _business_profile: business_profile::BusinessProfileNew,
+        business_profile: business_profile::BusinessProfileNew,
     ) -> CustomResult<business_profile::BusinessProfile, errors::StorageError> {
-        Err(errors::StorageError::MockDbError)?
+        let mut business_profiles = self.business_profiles.lock().await;
+
+        let business_profile_insert = business_profile::BusinessProfile {
```

Comment:
Quick question, could this be changed to the following?

```rust
let business_profile_insert = business_profile::BusinessProfile {
    ..business_profile
};
```

Reply:
I'd expect it to not work, using the struct update syntax needs both the source and destination structs to be the same type.

Reply:
Generally conversions between types is achieved by `From` and `Into` traits. Refer [this](https://doc.rust-lang.org/rust-by-example/conversion/from_into.html).

Reply:
Changed to use `impl From<BusinessProfileNew> for BusinessProfile`